src/main/java/teammates/logic/core/NotificationLogic.java

```diff
@@ -0,0 +1,55 @@
+package teammates.logic.core;
+
+import teammates.common.datatransfer.attributes.NotificationAttributes;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.storage.api.NotificationDb;
+
+/**
+ * Handles the logic related to notifications.
+ */
+public final class NotificationLogic {
+
+    private static final NotificationLogic instance = new NotificationLogic();
+
+    private final NotificationDb notificationDb = NotificationDb.inst();
+
+    private NotificationLogic() {
+        // prevent initialization
+    }
+
+    public static NotificationLogic inst() {
+        return instance;
+    }
+
+    /**
+     * Gets notification associated with the {@code notificationId}.
+     *
+     * @return null if no match found.
+     */
+    public NotificationAttributes getNotification(String notificationId) {
+        return notificationDb.getNotification(notificationId);
+    }
+
+    /**
+     * Updates/Creates the profile using {@link StudentProfileAttributes.UpdateOptions}.
```

Comment:
```suggestion
     * Updates/Creates the notification using {@link NotificationAttributes.UpdateOptions}.
```
Also, should we separate creating and updating the notification into two different functions? Doing so may be beneficial down the line if we were to add checks that apply only when updating the notification (e.g. disabling edit of title/message). It also doesn't really seem to make sense to throw an `EntityDoesNotExistExistException` here if we were to have the function handle both creating and updating of notification.

Reply:
I think the validations might be handled by the APIs when they receive relevant parameters as here is about the logic?

Reply:
Thanks for the comment. This API is for update only. I forgot to put the create one inside. Will do so in next commit.
Yes, they are separate.